backend/dao/winesDAO.js

```diff
@@ -94,29 +140,71 @@ module.exports = class WinesDAO {
       const displayCursor = cursor
         .limit(winesPerPage)
         .skip(winesPerPage * page);
-      //console.log(displayCursor);
+
       try {
         const winesList = await displayCursor.toArray();
-        //console.log(winesList);
-        //!console.log("wine list: ", winesList);
+
         const totalWines = await wines.countDocuments(query);
         console.log("total wines:", totalWines);
+        //*tblshooting, empty the query here..
+        //todo: test more usecases for problems
+        query = {};
+
         return { winesList, totalWines };
       } catch (e) {
         console.error(`Unable to convert cursor to array: ${e}`);
         return { wineList: [], totalWines: 0 };
       }
     }
   }
-
-  static async addWine(req, res, next) {
+  //? ************************************* */
+  //* ************Add Wine***************** */
+  //* ************************************* */
+  static async addWine(req) {
     try {
       const wineDoc = req;
-      //console.log(wineDoc);
 
       return await wines.insertOne(wineDoc);
     } catch (e) {
       console.error(`Unabe to post wine: ${e}`);
     }
   }
+  //? ************************************* */
+  //* ************Single Wine************** */
+  //* ************************************* */
+  static async getSingleWine({
+    filter = null,
+    page = 0,
+    winesPerPage = 5,
+  } = {}) {
+    console.log("filter: ", filter);
+    if (filter) {
+      query.wine_id = { $eq: filter.id };
+    }
+
+    console.log("query:", query);
+    let cursor;
+    try {
+      cursor = await wines.find(query);
+    } catch (e) {
+      console.error(`Unable to issue find query: ${e}`);
+      return { singleWine: 0 };

```

Comment:
This doesn't match the other values returned, here we return `singleWine` and in the other return statements we return `winesList, totalWines`.

Reply:
Oh yes, I had it set up originally for one wine, but could not have findById function... I want to change the whole function to be findById, wanted to push this one so front end could test it. I don’t have my computer until Monday, but will start working in it again, and change the whole function to fit findById

Reply:
I'd suggest to grab the first wine of the list in case there multiple wines.
It's better to define the future API of the function even it's not complete because people will start to use the function, and once you fix the issue and decide to change the value returned by the function, you'll have to change every place the function is called, besides being more work, this is prone to bugs.